routers/api/packages/nuget/auth.go

```diff
@@ -0,0 +1,45 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package nuget
+
+import (
+	"net/http"
+
+	"code.gitea.io/gitea/models"
+	user_model "code.gitea.io/gitea/models/user"
+	"code.gitea.io/gitea/modules/log"
+	"code.gitea.io/gitea/modules/timeutil"
+	"code.gitea.io/gitea/services/auth"
+)
+
+type Auth struct{}
+
+func (a *Auth) Name() string {
+	return "nuget"
+}
+
+// https://docs.microsoft.com/en-us/nuget/api/package-publish-resource#request-parameters
+func (a *Auth) Verify(req *http.Request, w http.ResponseWriter, store auth.DataStore, sess auth.SessionStore) *user_model.User {
+	token, err := models.GetAccessTokenBySHA(req.Header.Get("X-Nuget-Apikey"))
```

Comment:
Shouldn't it be
```suggestion
	token, err := models.GetAccessTokenBySHA(req.Header.Get("X-NuGet-ApiKey"))
```
according to the linked doc?

Reply:
Or should we use the current one as the headers are already case insensitive as per RFC?

Reply:
Good catch. I logged, what was present in the request:
![grafik](https://user-images.githubusercontent.com/1666336/183576086-9d67606c-0611-4538-8070-ca5ca0c19e34.png)

According to the source, the NuGet client sends what's stated in the docs but Go transforms the header keys again:
https://cs.opensource.google/go/go/+/refs/tags/go1.19:src/net/http/header.go;l=233-239

So I would just leave it like it is.